src/packages/utils/src/things/json-rpc/json-rpc-quantity.ts

```diff
@@ -115,6 +117,77 @@ export class Quantity extends BaseJsonRpcType {
     return firstNonZeroByte;
   }
 
+  private static getArgumentAsBigInt(argument: JsonRpcInputArg | Quantity): bigint {
+    const argumentType = typeof argument;
+    let argumentValue: bigint;
+
+    if (argumentType === "number") {
+      const argumentNumber = argument as number;
+      if (argumentNumber < 0) {
+        throw new Error("Cannot wrap a negative value as a json-rpc type.");
+      }
+      if (argumentNumber % 1) {
+        throw new Error("Cannot wrap a decimal as a json-rpc type.");
+      }
+      if (!isFinite(argumentNumber)) {
+        throw new Error(`Cannot wrap ${argument} as a json-rpc type.`);
+      }
+      argumentValue = BigInt(argumentNumber);
+    }  else if (argumentType === "bigint") {
+      if (argument < 0n) {
+        throw new Error("Cannot wrap a negative number as a JSON-RPC type.");
+      }
+      argumentValue = argument as bigint;
+    } else if (argumentType === "string") {
+      if ((argument as string).slice(2).toLowerCase() !== "0x") {
```

Comment:
We rely on `BigInt(value)`'s validation here. We have to explicitly check the first two chars, otherwise it may parse as decimal. I've added a few more test cases: https://github.com/trufflesuite/ganache/pull/3164/commits/e5b747bc9e6abb3858b2d098c7afd14bfb916ee8

Reply:
Ahh there already were tests with invalid hex, so you probably didn't need to add those. My miss!